ravenframework/Models/PostProcessors/Validations/Representativity.py

```diff
@@ -0,0 +1,489 @@
+          self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+        else:
+          pivotParameter = self.pivotParameter
+    evaluation ={k: np.atleast_1d(val) for k, val in  self._evaluate(dataSets, **{'dataobjectNames': names}).items()}
+
+    ## TODO: This is a placeholder to remember the time dependent case
+    # if pivotParameter:
+    #   # Uncomment this to cause crash: print(dataSets[0], pivotParameter)
+    #   if len(dataSets[0][pivotParameter]) != len(list(evaluation.values())[0]):
+    #     self.raiseAnError(RuntimeError, "The pivotParameter value '{}' has size '{}' and validation output has size '{}'".format( len(dataSets[0][self.pivotParameter]), len(evaluation.values()[0])))
+    #   if pivotParameter not in evaluation:
+    #     evaluation[pivotParameter] = dataSets[0][pivotParameter]
+    return evaluation
+
+  def _evaluate(self, datasets, **kwargs):
+    """
+      Main method to "do what you do".
+      @ In, datasets, list, list of datasets (data1,data2,etc.) to used.
+      @ In, kwargs, dict, keyword arguments
+      @ Out, outputDict, dict, dictionary containing the results {"feat"_"target"_"metric_name":value}
+    """
+    # # ## Analysis:
+    # # 1. Compute mean and variance:
+    # For mock model
+    self._computeMoments(datasets[0], self.featureParameters, self.features)
+    measurableNames = [s.split("|")[-1] for s in self.features]
+    measurables = [datasets[0][var].meanValue for var in measurableNames]
+    # For target model
+    self._computeMoments(datasets[1], self.targetParameters, self.targets)
+    FOMNames = [s.split("|")[-1] for s in self.targets]
+    FOMs = np.atleast_2d([datasets[1][var].meanValue for var in FOMNames]).reshape(-1,1)
+    # # 2. Propagate error from parameters to experiment and target outputs.
+    # For mock model
+    self._computeErrors(datasets[0],self.featureParameters, self.features)
+    measurableErrorNames = ['err_' + s.split("|")[-1] for s in self.features]
+    FOMErrorNames = ['err_' + s.split("|")[-1] for s in self.targets]
+    self._computeMoments(datasets[0], measurableErrorNames, measurableErrorNames)
+    UMeasurables = np.atleast_2d([datasets[0][var].meanValue for var in measurableErrorNames]).reshape(-1,1)
+    # For target model
+    self._computeErrors(datasets[1],self.targetParameters, self.targets)
+    self._computeMoments(datasets[1], FOMErrorNames, FOMErrorNames)
+    UFOMs = np.atleast_2d([datasets[1][var].meanValue for var in FOMErrorNames]).reshape(-1,1)
+    # # 3. Compute mean and variance in the error space:
+    self._computeMoments(datasets[0],['err_' + s.split("|")[-1] for s in self.featureParameters],['err_' + s2.split("|")[-1] for s2 in self.features])
+    self._computeMoments(datasets[1],['err_' + s.split("|")[-1] for s in self.targetParameters],['err_' + s2.split("|")[-1] for s2 in self.targets])
```

Comment:
computeErrors and computeMoments have been called, but no return is recorded. Could you explain?

Reply:
It adds them to self